internal/repositories/account/account.go

```diff
@@ -0,0 +1,264 @@
+package account
+
+import (
+	"context"
+
+	gmodel "github.com/DIMO-Network/identity-api/graph/model"
+	"github.com/DIMO-Network/identity-api/internal/helpers"
+	"github.com/DIMO-Network/identity-api/internal/repositories/base"
+	"github.com/DIMO-Network/identity-api/models"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/vektah/gqlparser/v2/gqlerror"
+	"github.com/volatiletech/sqlboiler/v4/queries/qm"
+	"golang.org/x/exp/slices"
+)
+
+// AccountPrefix is the prefix for a global token id for Accounts.
+const AccountPrefix = "Acct"
+
+type Repository struct {
+	*base.Repository
+}
+
+type AccountCursor struct {
+	OwnerAddress []byte
+	Kernel       []byte
+}
+
+var accountCursorColumnsTuple = "(" + models.KernelAccountColumns.OwnerAddress + ", " + models.KernelAccountColumns.Kernel + ")"
+
+func (r *Repository) GetAccount(ctx context.Context, by gmodel.AccountBy) (*gmodel.Account, error) {
+	if by.Signer != nil && by.Kernel == nil && by.Address == nil {
+		return r.GetAccountBySigner(ctx, *by.Signer), nil
+	}
+
+	if by.Kernel != nil && by.Signer == nil && by.Address == nil {
+		return r.GetAccountByKernel(ctx, *by.Kernel), nil
+	}
+
+	if by.Address != nil && by.Signer == nil && by.Kernel == nil {
+		return r.GetAccountAddress(ctx, *by.Address), nil
+	}
+
+	return nil, gqlerror.Errorf("Provide exactly one of `signer`, `kernel` or `address`.")
+}
+
+func (r *Repository) GetAccountBySigner(ctx context.Context, signer common.Address) *gmodel.Account {
+	all, err := models.KernelAccounts(models.KernelAccountWhere.OwnerAddress.EQ(signer.Bytes())).All(ctx, r.PDB.DBS().Reader)
+	if err != nil {
+		return nil
+	}
+
+	apiAccount := &gmodel.Account{
+		Signer: common.BytesToAddress(signer.Bytes()),
+	}
+
+	for _, acct := range all {
+		apiAccount.Kernel = append(apiAccount.Kernel, &gmodel.Kernel{
+			Address:   common.BytesToAddress(acct.Kernel),
+			CreatedAt: acct.CreatedAt,
+			Signer: &gmodel.Signer{
+				Address:     common.BytesToAddress(signer.Bytes()),
+				SignerAdded: acct.SignerAdded,
+			},
+		})
+	}
+
+	return apiAccount
+}
+
+func (r *Repository) GetAccountByKernel(ctx context.Context, kernel common.Address) *gmodel.Account {
```

Comment:
nit: These three functions could combined be a single function that takes in a `qm.QueryMod`